SpatialGDK/Source/SpatialGDK/Public/Interop/Connection/SpatialTraceEventDataBuilder.h

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Interop/Connection/SpatialTraceUniqueId.h"
+#include "SpatialCommonTypes.h"
+#include <WorkerSDK/improbable/c_trace.h>
+#include <WorkerSDK/improbable/c_worker.h>
+
+#define GDK_EVENT_NAMESPACE "unreal_gdk."
+
+#define PUSH_RPC_EVENT_NAME GDK_EVENT_NAMESPACE "push_rpc"
```

Comment:
We pass these 'static' strings to functions that do dynamic allocations for temporaries (as either FString or char*) why not have a function that takes an enum and returns you the correct static string, could maybe even be handled as constexpr function reducing overhead even further.

Reply:
Are you are confusing USpatialEventTracerUserInterface::TraceEvent and SpatialEventTracer::TraceEvent?

Reply:
Both take in strings?

Reply:
I am not sure what you are getting at, could you give some more context?

Reply:
Sorry for my poor clarity. I don't think we ever pass a 'static' string (i.e. PUSH_RPC_EVENT_NAME) into functions that take an FString so we should be good there. That's why I thought you might be referencing USpatialEventTracerUserInterface::TraceEvent which does take an FString. In the case of SpatialEventTracer::TraceEvent, is your suggestion for SpatialEventTracer::TraceEvent to take an enum and then to internally get the event type using the enum and give the result directly to the creation of Trace_Event?

Reply:
`take an enum and then to internally get the event type using the enum and give the result directly to the creation of Trace_Event` This

Reply:
Although I think you could do a template parameter too which would reduce the runtime cost even further.

Reply:
Sorry for my poor clarity. <- No worries happens to us all, just trying to understand the code : )

Reply:
This will require overloading the function as we will not know all event types at runtime. Users have the ability to specify their  own. We could tell users to add to the enum but that seems like it could be a step too far. 

Just for my understanding, we are saving here by only allocating a byte for the enum value rather than a pointer to the string?

Reply:
Yeah in the case of `const char*` the difference isnt that big. I am trying to think of ways of removing as many string ops as possible, the template suggestion would still save on having any parameter, but maybe this is getting a bit out of scope, I don't really have a handle on the performance requirements.

Reply:
The template solution would basically be baking in string literal in all the function permutations.